### ISSUE TITLE
xds_to_table fails when adding a column in the presence of an unknown field on the xds

### ISSUE THREAD

**Issue by JSKenyon** [ratt-ru/dask-ms#118]
* dask-ms version: master
* Python version: 3.6.9
* Operating System: Pop!_OS 18.04 LTS

### Description

I was attempting to write a new `BITFLAG` column to a dataset. It fails due to the presence of `CHAN_FREQ` on the xds even though `CHAN_FREQ` is not being written to disk. In my opinion, `dask-ms` should not attempt to figure things out for fields it isn't supposed to write.

### What I Did
```python
from daskms import xds_from_ms, xds_to_table, xds_from_table
import dask.array as da
import numpy as np

ms_name = "/path/to/MS"


if __name__ == "__main__":

    xds_list = xds_from_ms(ms_name,
                           index_cols=("TIME",),
                           group_cols=("SCAN_NUMBER",
                                       "FIELD_ID",
                                       "DATA_DESC_ID"),)

    spw_xds = xds_from_table(
        ms_name + "::SPECTRAL_WINDOW",
        columns="CHAN_FREQ",
        chunks={"row": 1, "chan": -1})[0]

    xds_list = [xds.assign({"BITFLAG": (("row", "chan", "corr"),
                            da.zeros_like(xds.DATA.data, dtype=np.uint32))})
                for xds in xds_list]

    xds_list = [xds.assign({"CHAN_FREQ": (("chan",),
                            spw_xds.CHAN_FREQ.data[0])})
                for xds in xds_list]

    write_xds_list = xds_to_table(xds_list, ms_name,
                                  columns=["BITFLAG"])
```

### Traceback
```
Traceback (most recent call last):
  File "/v2env/lib/python3.6/site-packages/daskms/descriptors/ms.py", line 98, in descriptor
    desc[k] = default_desc[k]
KeyError: 'CHAN_FREQ'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "extra_field.py", line 30, in <module>
    columns=["BITFLAG"])
  File "/v2env/lib/python3.6/site-packages/daskms/dask_ms.py", line 118, in xds_to_table
    table_proxy=table_proxy)
  File "/v2env/lib/python3.6/site-packages/daskms/writes.py", line 642, in write_datasets
    tp = _updated_table(table, datasets, columns, descriptor)
  File "/v2env/lib/python3.6/site-packages/daskms/writes.py", line 251, in _updated_table
    table_desc = builder.descriptor(variables, default_desc)
  File "/v2env/lib/python3.6/site-packages/daskms/descriptors/ms.py", line 100, in descriptor
    desc[k] = self.variable_descriptor(k, lv)
  File "/v2env/lib/python3.6/site-packages/daskms/descriptors/builder.py", line 44, in variable_descriptor
    return variable_column_descriptor(column, variable)
  File "/v2env/lib/python3.6/site-packages/daskms/descriptors/builder.py", line 146, in variable_column_descriptor
    ndims.add(v.dim)
AttributeError: 'Variable' object has no attribute 'dim'
```

**Comment by JSKenyon** [comment]
My current work-around is:
```python
xds_list = [xds.drop(["CHAN_FREQ"]) for xds in xds_list]
```
before the `xds_to_table` call.

**Comment by JSKenyon** [comment]
This is now partially fixed as a result of #200. Schemas are no longer created for dataset components which are not going to be written to disk. The above example will now run through, but will fail if `columns=["BITFLAG", "CHAN_FREQ"]`. This is a sensible failure case (writing a non-row column to the main table), although I think the error message could be more informative.
